### PR TITLE
Changed the complete portfolio as compared to week 1

### PR DESCRIPTION
**Pull request by ananyaativ**
I have completely changed the portfolio as compared to week 1.

### CODE REVIEW

**Comment by danthev** [comment]
This PR is based on the code before #2 was merged. Can you merge master (`git merge origin/master` while you're on the AnanyaDevelopment branch) and resolve the conflicts? That way I know which code is meant to be new.

**Comment by ananyaativ** [comment]
> This PR is based on the code before #2 was merged. Can you merge master (`git merge origin/master` while you're on the AnanyaDevelopment branch) and resolve the conflicts? That way I know which code is meant to be new.

I have resolved the conflicts. Let me know if I need to do anything else.

**Comment by danthev** [comment]
Thanks Ananya. It looks like you still you need to push the changes to Github (`git push`), unless it accidentally went to another branch.

**Comment by ananyaativ** [comment]
> Thanks Ananya. It looks like you still you need to push the changes to Github (`git push`), unless it accidentally went to another branch.

Oh yeah I forgot to do that. It's done now. Sorry about that

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java [hunk 1]
```diff
@@ -5,14 +5,26 @@
...
 
 /** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
 @WebServlet("/hello")
 public class HelloWorldServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<String> facts=new ArrayList<>();
```
Comment (danthev) [inline portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:17]: Optional: 
I like to use [ImmutableList.of](https://guava.dev/releases/20.0/api/docs/com/google/common/collect/ImmutableList.html#of-E-E-E-E-E-E-E-E-E-E-E-E-E...-) for these kind of constants, though it requires Guava. [`Arrays.asList("foo", "bar")`](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList-T...-) is another alternative that's pretty close to a list literal.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,12 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+jQuery(document).ready(function($) {
+
+
+	var mastheadheight = $('.ds-header').outerHeight();
+    console.log(mastheadheight);
```
Comment (danthev) [inline portfolio/src/main/webapp/script.js:19]: This is probably not needed (or at least not indented correctly).

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,12 +12,46 @@
...
+	}).scroll();
+
+
+ });
+
+ /**
+  * fetches the string from servlet
+  */
+
+  async function getString(){
```
Comment (danthev) [inline portfolio/src/main/webapp/script.js:38]: Can you use something more descriptive? Like `loadQuote` or `fetchAndRenderQuote()`.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,12 +12,46 @@
...
+      const responseFromServer= await fetch('/hello');
+      const textFromResponse= await responseFromServer.json();
```
Comment (danthev) [inline portfolio/src/main/webapp/script.js:40]: I'd recommend running a linter to format everything, `textFromResponse` should have a space right after.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,12 +12,46 @@
...
+     // console.log(textFromResponse);
```
Comment (danthev) [inline portfolio/src/main/webapp/script.js:41]: This can go.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -12,12 +12,46 @@
...
+
+
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
+function addRandomQuote() {
   const quotes =
-      ['I am unbeatable in Othello', 'It’s the little things in life', 'Fake it till you make it' , 'I have my second toe bigger than my first toe', 'I can touch my nose with my tongue'];
+      ['I have never lost a game of Othello.', 'I was born in Delhi, India.', 'My second toe is bigger than my first toe.', 'I can touch my nose with my tongue.'];
```
Comment (danthev) [inline portfolio/src/main/webapp/script.js:54]: This is something for a follow-up PR, but if you're fetching all quotes in `getString`, you can probably save those and just replace which element is being shown. That way you also don't need to update the list in two places.

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -1,83 +1,276 @@
...
+        </section>
+        <span class="ds-copyright">Copyright © Ananya Aggarwal </span>
+     </div>
+   </footer>
+
+
+
+    
+
+    <!-- Option 2: Separate Popper and Bootstrap JS -->
```
Comment (danthev) [inline portfolio/src/main/webapp/index.html:266]: This isn't needed I think. ;)

**Review by danthev — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (ananyaativ) [reply portfolio/src/main/webapp/script.js:40]: is there a way to add it as a plugin in the shell?

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (ananyaativ) [reply portfolio/src/main/webapp/script.js:54]: I removed the addRandomQuote method and now the getString fetches all the strings and generates a random string. But when I open the webpage I see the string already even when the button is not pressed and I am not sure why. All I want is to display the random string only when the button is pressed.

Comment (danthev) [reply portfolio/src/main/webapp/script.js:54]: I can't see your updated code, but could it be that your function is still mentioned under `<body onload="getString()">`? That will trigger as soon as the website is shown.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (danthev) [reply portfolio/src/main/webapp/script.js:40]: I think the easiest way is to format in the Cloud Editor, like this: image.

There's some settings you can change regarding formatting in the settings, if that's not enough you could try installing ESLint manually from the command line: https://eslint.org/docs/user-guide/getting-started.

**Pull request merged**